src/file.jl

```diff
@@ -684,17 +684,18 @@ Base.@propagate_inbounds function parserow(startpos, row, numwarnings, ctx::Cont
                 # extra columns on this row, let's widen
                 ctx.silencewarnings || toomanycolumns(ncols, rowoffset + row)
                 j = i + 1
-                T = ctx.streaming ? Union{ctx.stringtype, Missing} : NeedsTypeDetection
                 while pos <= len && !Parsers.newline(code)
-                    col = Column(T, ctx.options)
+                    col = initialize_column(j, ctx)
                     col.anymissing = ctx.streaming || rowoffset == 0 && row > 1 # assume all previous rows were missing
                     col.pool = ctx.pool
+                    # TODO: Do we need to check `nonstandardtype(T)` and potentially create a `Context` an updated `customtypes`?
```

Comment:
thoughts, @quinnj ?

Reply:
Oof, this is rough. This is exposing a big design issue w/ how we're currently doing parsing. And it's not obvious how to fix. Unfortunately just creating a new context isn't quite enough because we're cheating a little and passing `ctx.customtypes` to `parsechunk!` so that it gets specialized. So we would need to recall `parsechunk` _and_ pass the new customtypes, and my brain can't quite wrap around how this would affect the multithreaded context we're in.

BUT, I think this fix is good for now, but we maybe just need to make a note that non-standard types aren't supported on widened columns.

Reply:
ah, yeah this is complex -- i agree it's not obvious how to support it and it does seem _very_ niche.

I've added an `@error` log to inform us if we are ever in this scenario and tell a user parsing will probably fail
https://github.com/JuliaData/CSV.jl/pull/1023/commits/e00fe5e9fcd26c7076280ad7b69e2d519040012f